scripts/bond-calculator.py

```diff
@@ -0,0 +1,128 @@
+or an amount in either satoshi or bitcoin (e.g., 150000, 0.1, 10.123, 10btc),
+calculate fidelity bond values for all possible locktimes in a one-year period
+(12 months, you can change that with the `-m --months` option).
+By default it uses the values from your joinmarket.cfg,
+you can override these with the `-i --interest` and `-e --exponent` options.
+Additionally, you can export the orderbook from ob-watcher.py and use the data here
+with the `-o --orderbook` option, this will compare the results from this script
+with the fidelity bonds in the orderbook.
+"""
+
+log = get_log()
+
+
+def main() -> None:
+    parser = OptionParser(
+        usage="usage: %prog [options] UTXO or amount",
+        description=DESCRIPTION,
+    )
+    add_base_options(parser)
+    parser.add_option(
+        "-i",
+        "--interest",
+        action="store",
+        type="float",
+        dest="interest",
+        help="Interest rate to use for fidelity bond calculation (instead of interest_rate config)",
+    )
+    parser.add_option(
+        "-e",
+        "--exponent",
+        action="store",
+        type="float",
+        dest="exponent",
+        help="Exponent to use for fidelity bond calculation (instead of bond_value_exponent config)",
+    )
+    parser.add_option(
+        "-m",
+        "--months",
+        action="store",
+        type="int",
+        dest="months",
+        help="For how many months to calculate the fidelity bond values, each month has its own stats (default 12)",
+        default=12,
+    )
+    parser.add_option(
+        "-o",
+        "--orderbook",
+        action="store",
+        type="str",
+        dest="path_to_json",
+        help="Path to the exported orderbook in JSON format",
+    )
+
+    options, args = parser.parse_args()
+    load_program_config(config_path=options.datadir)
+    if len(args) != 1:
+        log.error("Invalid arguments, see --help")
+        sys.exit(EXIT_ARGERROR)
+    if options.path_to_json:
+        try:
+            with open(options.path_to_json, "r", encoding="UTF-8") as orderbook:
+                orderbook = loads(orderbook.read())
+        except FileNotFoundError as exc:
+            log.error(exc)
+            sys.exit(EXIT_ARGERROR)
+    else:
+        orderbook = None
+    try:
+        amount = amount_to_sat(args[0])
+        confirm_time = None
+    except ValueError:
+        # If it's not a valid amount then it has to be a UTXO
+        if jm_single().bc_interface is None:
+            log.error("For calculation based on UTXO access to Bitcoin Core is required")
+            sys.exit(EXIT_FAILURE)
+        success, utxo = utxostr_to_utxo(args[0])
+        if not success:
+            # utxo contains the error message
+            log.error(utxo)
+            sys.exit(EXIT_ARGERROR)
+        utxo_data = jm_single().bc_interface.query_utxo_set(utxo, includeconfs=True)[0]
+        amount = utxo_data["value"]
+        if utxo_data["confirms"] <= 0:
```

Comment:
Here you'd want to treat the `<0` case separately, if conflicted i.e. invalid don't treat it the same as unconfirmed. (Actually I rarely think much about this case, since it happens basically never, but see e.g. [here](https://bitcoin.org/en/release/v0.12.0#wallet-negative-confirmations-and-conflict-detection) and output of `bitcoin-cli help gettransaction` (same applies to `gettxout` afaik though the help does not detail it).

Reply:
Done in 6f68498dbf38fa2be4a2f5b298b49a548dfbe877

Now < 0 confirms is considered a failure (maybe a warning was enough, not sure how useful it is to calculate value of an invalid UTXO).
== 0 confirms triggers a warning instead.